bin/tests/integration/config_tests.rs

```diff
@@ -50,19 +50,25 @@ fn test_read_config() {
     assert_eq!(config.directory(), Path::new("/var/named"));
 
     assert_eq!(config.zones()[0].zone, "localhost");
-    assert_eq!(config.zones()[0].zone_type, ZoneType::Primary);
+    assert_eq!(config.zones()[0].zone_type(), ZoneType::Primary);
     assert_eq!(
         config.zones()[0]
+            .zone_type_config
```

Comment:
Nit: let's add a helper like `zone_server(i: usize) -> &ServerStoreConfig` to reduce the boilerplate a little?